### PR TITLE
add method mapping and tests

### PR DESCRIPTION
**Pull request by eclendenning-va**

### CODE REVIEW

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/observation/MethodMapping.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+public class MethodMapping {
+
+  /** Converts qualifier vuids into codings for the Method CodeableConcept. */
+  public static Coding toCoding(VistaQualifiers qualifier) {
+    if (qualifier == null) {
+      return null;
+    }
+    var coding = Coding.builder().system("http://snomed.info/sct");
+    switch (qualifier) {
```
Comment (bschofield-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/observation/MethodMapping.java:21]: remove the case by making the enum smarter

File: vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/controller/observation/MethodMappingTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import java.util.List;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class MethodMappingTest {
+  static Stream<Arguments> toMethod() {
+    return Stream.of(
+        arguments("4711345", "258104002"),
```
Comment (bschofield-va) [inline vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/controller/observation/MethodMappingTest.java:16]: verified code mappings match spreadsheet attached to story

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/observation/MethodMapping.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            .collect(Collectors.toList());
+    if (coding.isEmpty()) {
+      return null;
+    }
+    return CodeableConcept.builder().coding(coding).build();
+  }
+
+  @AllArgsConstructor
+  enum VistaQualifiers {
+    ACTUAL("4711345", "258104002", "Measured (qualifier value)"),
```
Comment (bschofield-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/observation/MethodMapping.java:44]: verified descriptions/codes match per spreadsheet attached to story

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/observation/MethodMapping.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+public class MethodMapping {
+
+  /** Converts qualifier vuids into codings for the Method CodeableConcept. */
+  public static Coding toCoding(VistaQualifiers qualifier) {
+    if (qualifier == null) {
+      return null;
+    }
+    return Coding.builder()
+        .system("http://snomed.info/sct")
+        .display(qualifier.display())
```
Comment (bschofield-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/observation/MethodMapping.java:22]: nice!

**Review by bschofield-va — APPROVED** [review]

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/observation/VistaVitalToR4ObservationTransformer.java [hunk 5]
```diff
@@ -131,9 +131,17 @@ String idFrom(String id) {
     return toResourceId(patientIcn, vistaSiteId, VprGetPatientData.Domains.vitals, id);
   }
 
+  CodeableConcept method(Vitals.Measurement measurement) {
+    if (isBlank(measurement) || isEmpty(measurement.qualifiers())) {
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/observation/VistaVitalToR4ObservationTransformer.java:135]: `isEmpty(measurement.qualifiers())` -> `isBlank(measurement.qualifiers())`

isBlank performs a null check as well as an empty check.

File: vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/controller/observation/VistaVitalToR4ObservationTransformerTest.java [hunk 6]
```diff
@@ -14,29 +14,85 @@
...
+                        Coding.builder()
+                            .system("http://snomed.info/sct")
+                            .code("258104002")
+                            .display("Measured (qualifier value)")
+                            .build()))
+                .build());
+  }
+
+  @Test
+  void methodWithUnknownVuidsReturnsNull() {
```
Comment (jhulbert-va) [inline vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/controller/observation/VistaVitalToR4ObservationTransformerTest.java:62]: One bad boi and one good boi?

**Review by jhulbert-va — COMMENTED** [review]
Should we add a qualifier into the samples class?

**Review by jhulbert-va — APPROVED** [review]

**Pull request merged**